src/ipc_listener.cpp

```diff
@@ -208,9 +260,13 @@ int zmq::ipc_listener_t::set_address (const char *addr_)
 
 int zmq::ipc_listener_t::close ()
 {
-    zmq_assert (s != retired_fd);
-    int rc = ::close (s);
-    errno_assert (rc == 0);
+    int rc;
+
+    if ( s != -1 ) {
+      zmq_assert (s != retired_fd);
```

Comment:
Yes, and that's why, I think, the code as-is did a return -1 instead of goto err if open_socket failed.

We need the assert to enforce that this API is not used improperly, IE: called on a socket that's already closed. Do you think it's possible to leave that in with your changes?